tensorflow_datasets/core/image_visualizer.py

```diff
@@ -0,0 +1,235 @@
+    return {
+        'label_key': label_key
+    }
+
+  def _plot_objects(self, plt, ax, ds_info, example, image_key, label_key):
+    """Plots objects for a single example"""
+    # Plot the label
+    if label_key:
+      label = example[label_key]
+      label_str = ds_info.features[label_key].int2str(label)
+      plt.xlabel("{} ({})".format(label_str, label))
+
+    return plt, ax
+
+  def match(self, ds_info):
+    """Checks if dataset belongs to image dataset category"""
+    image_keys = _extract_keys(ds_info, features_lib.Image)
+
+    if not image_keys:
+      return False
+    return True
+
+class ObjectVisulaizer(ImageVisualizer):
+  """Class for Object Detection Datasets"""
+
+  def _infer_object_keys(self, ds_info):
+    """Infers object keys to be passed to _plot_objects()"""
+    sequence_keys = _extract_keys(ds_info, features_lib.Sequence)
+
+    # Infer the BBox keys in the sequence feature connectors
+    bbox_keys = []
+    for key in sequence_keys:
+      if isinstance(ds_info.features[key].feature, features_lib.FeaturesDict):
+        bbox_keys = _extract_keys(ds_info, features_lib.BBoxFeature, key)
+        if len(bbox_keys) > 0:
+          sequence_key = key
+          break
+
+    # Infer the label keys in the sequence feature connector (BBox Labels)
+    label_keys = _extract_keys(ds_info, features_lib.ClassLabel,
+                               sequence_key)
+
+    # Taking first label key since some datasets have multiple label keys
+    # for BBoxes as in voc dataset
+    label_key = label_keys[0] if len(label_keys) > 0 else None
+    if not label_key:
+      logging.info("Was not able to auto-infer label.")
+    bbox_key = bbox_keys[0] if len(bbox_keys) > 0 else None
+
+    return {
+        'label_key': label_key,
+        'bbox_key': bbox_key,
+        'sequence_key': sequence_key
+    }
+
+  def _plot_objects(self, plt, ax, ds_info, example, image_key, label_key, \
+                    bbox_key, sequence_key):
+    """Plots objects for a single example"""
+    image = example[image_key]
+
+    bboxes = example[sequence_key][bbox_key]
+    height = image.shape[0]
+    width = image.shape[1]
+    font_size = int((11./3.) * self.plot_scale)
+
+    # Plot the label and bounding boxes
+    for idx, box in enumerate(bboxes):
+      bbox = _bb_hw(box, height, width)
+      _draw_rectangle(ax, bbox)
+      if label_key:
+        text = ds_info.features[sequence_key][label_key]\
+              .int2str(example[sequence_key][label_key][idx])
+        _draw_text(ax, bbox[:2], text, size=font_size)
+
+    return plt, ax
+
+  def match(self, ds_info):
+    """Checks if dataset belongs to object detection category"""
+    image_keys = _extract_keys(ds_info, features_lib.Image)
+
+    if not image_keys:
+      return False
+
+    sequence_keys = _extract_keys(ds_info, features_lib.Sequence)
+
+    bbox_keys = []
+    for key in sequence_keys:
+      if isinstance(ds_info.features[key].feature, features_lib.FeaturesDict):
```

Comment:
Rather than this loop, how about a generic

```python
sequence_key, bbox_key = _extract_nested_keys(
    ds_info, [features_lib.Sequence, features_lib.FeaturesDict, features_lib.BBoxFeature])
```
It would return a new key for each `FeatureDict`, including the top-level one.